Fixes #8260 Moved Support AnkiDroid to navigation drawer

mrudultora opened the pull request:
## Purpose / Description
Moved **Support AnkiDroid** to Navigation Drawer.

## Fixes
Fixes #8260 

## Approach
Created a new method and only used the **LinkItem** class in the array of **Item**.

Updated Screenshots :

![1](https://user-images.githubusercontent.com/65870389/113617664-3dac3200-9674-11eb-9a44-732ce2012363.jpg)

![2](https://user-images.githubusercontent.com/65870389/113617667-3f75f580-9674-11eb-979a-9f45ef61b4cd.jpg)


## How Has This Been Tested?
Tested on real device (Redmi Note 7S and API 29).

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

mrudultora commented:
@david-allison-1 I have not removed **Support AnkiDroid** from the Help Dialog. Should I remove that ?

david-allison-1 commented:
> @david-allison-1 I have not removed **Support AnkiDroid** from the Help Dialog. Should I remove that ?

I think you should

david-allison-1 commented:
Might want to remove the old value
```
  /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/values/01-core.xml:29: Error: Duplicate string value "Support AnkiDroid", used in help_title_support_ankidroid, support_ankidroid [DuplicateCrowdInStrings]
      <string name="support_ankidroid">Support AnkiDroid</string>
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/values/03-dialogs.xml:232: Duplicates value in support_ankidroid. Add a comment attribute to explain this duplication
```

david-allison-1 reviewed (commented):
Looks good

* Could you add a screenshot of the new nav bar in dark and light mode
* As mentioned, could you remove the section from the "Help" dialog

mrudultora commented:
@david-allison-1 Changes done. Uploaded screenshots in the PR template.

mrudultora commented:
I didn't made changes to this but still getting lint error.
```
  /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/res/drawable/ic_heart_black_24dp.xml:1: Error: The resource R.drawable.ic_heart_black_24dp appears to be unused [UnusedResources]
  <vector xmlns:android="http://schemas.android.com/apk/res/android"
  ^
```

david-allison-1 commented on AnkiDroid/src/main/res/menu/navigation_drawer.xml:
```diff
@@ -25,6 +25,10 @@
             android:id="@+id/nav_settings"
             android:icon="@drawable/ic_settings_black_24dp"
             android:title="@string/settings"/>
+        <item
```
I'd move this to the bottom

mrudultora replied:
Okay

david-allison-1 reviewed (changes requested):
You've removed a reference to the heart, so now it's unused - you can delete the file

One nitpick when it comes to UI (personal preference - feel free to disagree) and good to go.

Thanks!

I think we might have some debate here over the best icon to use

mrudultora commented:
@david-allison-1 I have added a test also. If the test is not needed I would remove that and also fixed the old test as the child items are now reduced to3 from 4.

mrudultora commented:
For icon, I agree that everyone's view would be required. This was the only one available on Android Studio, so I used this.

david-allison-1 reviewed (approved):
LGTM - might want to update the screenshots

mrudultora commented:
Yeah, I am updating that, just few seconds.

mikehardy reviewed (approved):
Nice! code looks really clean, this all works for me - thanks preserving the analytics hit on support opening, since we just added that and we're curious whether it's used a bunch (and how / when that happens) or not in order to improve it
